sdk/metric/metricdata/data.go

```diff
@@ -211,6 +213,14 @@ type Extrema[N int64 | float64] struct {
 	valid bool
 }
 
+// MarshalJson converts the Extrema value to JSON number
+func (e *Extrema[N]) MarshalJSON() ([]byte, error) {
```

Comment:
Implementing `MarshalText` will be more general and support more use cases. Can we implement that instead?

Reply:
I've started with `MarshalText` but moved to `MarshalJSON` to keep API consistent. Rest of the numerical values are represented as json `number`, which cannot be done with `MarshalText` as json encoder will translate it as `string`. 

Can we agree on keeping `MarshalJSON` and implementing `MarshalText` next to it? 
As per [json docs](https://pkg.go.dev/encoding/json#Marshal), if `MarshalJSON` is implemented it will be preferred over `MarshalText` for json encoding. So we'll have no issues with JSON types and `Extrema` type will still be represented in encoders that utilize `MarshalText`